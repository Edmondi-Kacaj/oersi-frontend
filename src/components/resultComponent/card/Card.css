.cardShow {
  height: 24%;
}

.card-img {
  width: 100% !important;
}

.stats {
  margin-left: 2%;
}

ul > li > label > span {
  margin-right: 5%;
}

.card-title {
  color: cadetblue;
}

.card-footer {
  padding: 0.75rem 1.25rem !important;
  background-color: rgba(0, 0, 0, 0.03) !important;
  border-top: 1px solid rgba(0, 0, 0, 0.125) !important;
}

.legend-span {
  white-space: unset;
  overflow: hidden;
  text-overflow: ellipsis;
}

.block-with-text {
  overflow: hidden;
  display: -webkit-box;
  -webkit-line-clamp: 5;
  -webkit-box-orient: vertical;
}

.block-with-text:active {
  display: block;
}
