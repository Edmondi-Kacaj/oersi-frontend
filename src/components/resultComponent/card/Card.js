import React from "react";
import moment from "moment";
import "./Card.css";

class Card extends React.Component {
  render() {
    return (
      <React.Fragment>
        <div className="col-md-12" key={Math.random()}>
          <div className="card">
            <div className="">
              <a href={this.props.url}>
                <h4
                  data-toggle="tooltip"
                  data-placement="left"
                  title="Tooltip on left"
                  className="title text-center "
                >
                  {this.props.name}
                </h4>
              </a>
              <hr />
              <div className="row">
                <div className="col-md-10">
                  <div className="row">
                    <div className="col-md-3"></div>
                    <div className="col-md-5">
                      <p className=" title text-center">
                        <b>Author : </b> {this.props.author}
                      </p>
                    </div>
                    <div className="col-md-4">
                      <p className="category text-center">
                        License : {this.props.license}
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div className="content">
              <div className="col-md-12">
                <div className="row no-gutters">
                  <div className="col-md-4">
                    <a href={this.props.url}>
                      <img
                        src={this.props.thumbnail}
                        className="card-img"
                        alt={this.props.name}
                      />
                    </a>
                  </div>
                  <div className="col-md-8">
                    <div className="card-body">
                      {/* <a href={this.props.url}>
                        <h5 className="card-title">{this.props.name}</h5>
                      </a> */}
                      <p className="card-text block-with-text">
                        {this.props.about}
                      </p>
                      <hr></hr>
                      <p className="card-text">
                        <small className="text-muted">
                          {moment(this.props.timestamp).format("MMM Do YY")}
                        </small>
                      </p>
                    </div>
                  </div>
                </div>
              </div>
              <div className="footer card-footer font-size-responsive">
                <div className="col-md-12 col-sm-12 col-lg-12">
                  <div className="row">
                    <div className="legend">
                      {this.props.tags.split(/,| /).map(item => {
                        return (
                          <span
                            key={Math.random()}
                            className="badge badge-pill badge-info"
                          >
                            {item}
                          </span>
                        );
                      })}
                    </div>
                  </div>
                </div>
                <hr />
                <div className="stats">
                  <i className="fa fa-history"></i>
                  <strong> Last update : </strong>
                  {moment(this.props.modificationdate).format("YYYY.MM.DD")}
                </div>
                <div className="stats">
                  <i className="fa fa-file"></i> <strong> Type : </strong>
                  {this.props.learningresourcetype}
                </div>
                <div className="stats">
                  <i className="fa fa-language"></i>
                  <strong> Language : </strong>
                  {this.props.inlanguage}
                </div>
                <div className="stats">
                  <i className="fa fa-osi"></i>
                  <strong> Source : </strong>
                  {this.props.source}
                </div>
              </div>
            </div>
          </div>
        </div>
      </React.Fragment>
    );
  }
}

export default Card;
