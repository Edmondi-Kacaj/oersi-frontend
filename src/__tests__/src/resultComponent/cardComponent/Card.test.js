import React from "react";
import { mount, shallow } from "../../../../setupFiles";
import Card from "../../../../components/resultComponent/card/Card";

const dataPropsPassing = {
  author: "Edmond",
  license: "CC-LC-UU-Test",
  url: "http://test.com",
  thumbnail: "http://testphotourl.com",
  name: "Document Test",
  about: "This is just a object for test",
  timestamp: "2020-02-20T17:35:02.537Z",
  modificationdate: "2020-02-20T17:35:02.537Z",
  tags:
    "Sprachgebrauch , Sprachpraxis , Spracherhalt , Herkunftssprache(n) , Deutsch , Migration , Sprachenvielfalt , Mehrsprachigkeit , Bildung und Erziehung , Sprache"
};

describe("CardCompnent ==> Test UI  ", () => {
  let wrapperShadow;
  beforeEach(() => {
    wrapperShadow = shallow(<Card {...dataPropsPassing} />);
  });

  it("CardCompnent : should render correctly", () => {
    expect(wrapperShadow).toMatchSnapshot();
  });

  it(" CardCompnent:  should  wraps content in a div with .card class", () => {
    expect(wrapperShadow.find(".card").length).toBe(1);
  });
});

describe("CardCompnent  ==> Test Props of Component", () => {
  let wrapperMount;
  beforeEach(() => {
    wrapperMount = mount(<Card {...dataPropsPassing} />);
  });

  it(" CardCompnent:  should be not empty props ", () => {
    expect(wrapperMount.props()).toBeTruthy();
  });

  it("CardCompnent :  should be props same as Object ", () => {
    expect(wrapperMount.props()).toMatchObject(dataPropsPassing);
  });

  it("CardCompnent :  should be possibly to add props ", () => {
    wrapperMount.setProps({ cardProp: "Test" });
    expect(wrapperMount.prop("cardProp")).toEqual("Test");
  });
});
