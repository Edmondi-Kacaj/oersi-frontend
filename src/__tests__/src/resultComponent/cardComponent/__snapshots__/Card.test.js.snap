// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardCompnent ==> Test UI   CardCompnent : should render correctly 1`] = `ShallowWrapper {}`;
